### ISSUE TITLE
Will context data be overwritten？

### ISSUE THREAD

**Issue by congim** [sagikazarmark/go-gin-gorm-opencensus#6]
```go
func WithContext(ctx context.Context, db *gorm.DB) *gorm.DB {
	return db.Set(contextScopeKey, ctx)
}
```

go "func A " get *gorm.DB, and set context, at same time go "func B"  get *gorm.DB and set context.

it's will be overwritten?

**Comment by congim** [comment]
@sagikazarmark

**Comment by congim** [comment]
fix it
```go
// WithContext sets the current context in the db instance for instrumentation.
func WithContext(ctx context.Context, db *gorm.DB) *gorm.DB {
	return db.New().Set(contextScopeKey, ctx)
}
```

**Comment by sagikazarmark** [comment]
Fixed in #7
